application/libraries/Api/Rest/V1/SchemaFactory/SchemaFactorySurveyDetail.php

```diff
@@ -24,48 +25,42 @@ public function make(): Schema
             )
         );
 
-        $questionSchema = (new SchemaFactoryQuestion())->make();
-        $questionSchema = $questionSchema->properties(
-            ...array_merge(
-                $questionSchema->properties ?? [],
-                [$questionAttributesSchema, $answersSchema]
-            )
+        $questionSchema = (new SchemaFactoryQuestion())->make(
+            $answersSchema,
+            $questionAttributesSchema
         );
-        $questionsSchema = Schema::array('questions')->items(
-            AllOf::create()->schemas(
+        $questionsSchema = Schema::object('questions')->properties(
+            AllOf::create('0')->schemas(
```

Comment:
What's the '0' doing here?

Reply:
We replaced an array with an object keys on the id. This is a dummy id used in the schema.

Reply:
Oh ok. Maybe you can add a comment on that in the code? It was an eye-browse raiser for me. ^^

Reply:
Same as the other one. Its a dummy id for the question schema since we replaced array with an object indexed by id.